pkg/settings/doc.go

```diff
@@ -52,22 +74,41 @@ rate limit into a client or something, passing a `*FooSetting` rather than a
 `Foo` and waiting to call `.Get()` until the value is actually used ensures
 observing the latest value.
 
+# Changing names of settings
+
+Sometimes for UX or documentation purposes it is useful to rename a setting.
+However, to preserve backward-compatibility, its key cannot change.
+
+There are two situations possible:
+
+  - The setting currently has the same name as its key. To rename the setting,
+    use the `WithName()` option.
+
+  - The setting already has another name than its key. In this case,
+    modify the name in the `WithName()` option but also add a
+    `WithRetiredName()` option to redirect users to the new name.
```

Comment:
Probably obvious but you can say "also add a WithRetiredName() option with the previous name. A notice will direct users to the new name."